src/random.rs

```diff
@@ -82,6 +82,12 @@ fn additional_data() -> Result<[u8; 32], RandomnessError> {
 pub fn random_bytes() -> Result<[u8; 32], RandomnessError> {
     let mut engine = sha256::HashEngine::default();
 
+    #[cfg(not(any(target_arch = "x86", target_arch = "x86_64")))]
+    if let Some(bytes) = hardware_randomness()? {
+        engine.input(&bytes);
+    }
+
+    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
```

Comment:
Nice, i had forgotten this. I think this is actually what fixes your compilation on M1.